enterprise/coderd/coderd.go

```diff
@@ -450,11 +456,165 @@ func (api *API) updateEntitlements(ctx context.Context) error {
 		}
 	}
 
+	if changed, enabled := featureChanged(codersdk.FeatureWorkspaceProxy); changed {
+		if enabled {
+			fn := derpMapper(api.Logger, api.ProxyHealth)
+			api.AGPL.DERPMapper.Store(&fn)
+		} else {
+			api.AGPL.DERPMapper.Store(nil)
+		}
+	}
+
 	api.entitlements = entitlements
 
 	return nil
 }
 
+// getProxyDERPStartingRegionID returns the starting region ID that should be
+// used for workspace proxies. A proxy's actual region ID is the return value
+// from this function + it's RegionID field.
+//
+// Two ints are returned, the first is the starting region ID for proxies, and
+// the second is the maximum region ID that already exists in the DERP map.
+func getProxyDERPStartingRegionID(derpMap *tailcfg.DERPMap) (sID int, mID int) {
+	maxRegionID := 0
+	for _, region := range derpMap.Regions {
+		if region.RegionID > maxRegionID {
+			maxRegionID = region.RegionID
+		}
+	}
+	if maxRegionID < 0 {
+		maxRegionID = 0
+	}
```

Comment:
It could be less than 0?!

Reply:
I think region IDs can be less than zero there's no real bound for them AFAICT